src/github.com/couchbaselabs/sync_gateway/base/bucket.go

```diff
@@ -77,6 +85,23 @@ func (bucket couchbaseBucket) View(ddoc, name string, params map[string]interfac
 }
 
 func (bucket couchbaseBucket) StartTapFeed(args walrus.TapArguments) (walrus.TapFeed, error) {
+	// Use tap only when it's explicitly requested, or if DCP isn't supported
+	if bucket.spec.FeedType == "tap" {
```

Comment:
We're not actually switching on bucket.spec.FeedType here, though - we want to force tap if it's set, but users don't need to specify 'DCP' to use DCP.  I agree the control structure doesn't read terribly well (particularly with the multiple paths that end up calling StartCouchbaseTapFeed), but I don't know if an enum changes that.